documentation/jetty/modules/programming-guide/pages/arch/threads.adoc

```diff
@@ -166,7 +163,7 @@ The modes of task consumption discussed above are captured by the `org.eclipse.j
 * `org.eclipse.jetty.util.thread.strategy.ProduceExecuteConsume` for the `Produce-Execute-Consume` mode.
 * `org.eclipse.jetty.util.thread.strategy.ExecuteProduceConsume` for the `Execute-Produce-Consume` mode.
 
-There is an additional implementation of `ExecutionStrategy`, namely `org.eclipse.jetty.util.thread.strategy.AdaptiveExecutionStrategy`, that takes into account the `InvocationType` of the task when the task implements `Invocable` (otherwise it assumes `BLOCKING`).
+There is an additional implementation of `ExecutionStrategy`, namely `org.eclipse.jetty.util.thread.strategy.AdaptiveExecutionStrategy`, that heuristically chooses between the other strategies. The implementation takes into account both the availability of idle threads and the `InvocationType` of the task when choosing which specific strategy to apply.
```

Comment:
One line per sentence.

"availability of idle threads" is not correct: it does not matter if idle threads are available, there must be reserved threads available.

This concept is explained more precisely 2 paragraphs later, so I liked the original text better.

Reply:
The previous version was wrong as it left out half the basis of the adaption.   I've modified both this paragraph and the one 2 later so they are better linked and the later expands on this one.